test/e2e/suite/internal/client/client.go

```diff
@@ -41,19 +41,22 @@ var (
 	certmanagerClientLog = log.RegisterScope("certmanagerclient", "cert-manager client debugging")
 )
 
+var _ security.Client = &certmanagerClient{}
+
 type certmanagerClient struct {
 	caEndpoint    string
 	enableTLS     bool
 	caTLSRootCert []byte
 	clusterID     string
 	token         string
+	certificate   tls.Certificate
 
 	client securityapi.IstioCertificateServiceClient
 	conn   *grpc.ClientConn
 }
 
 // NewCertManagerClient create a CA client for cert-manager istio agent.
-func NewCertManagerClient(endpoint, token string, tls bool, rootCert []byte, clusterID string) (security.Client, error) {
```

Comment:
note: I agree with this change - https://go.dev/wiki/CodeReviewComments#interfaces